apps/web/src/app/_components/search/company-search-bar.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState } from "react";
+import { useFormContext } from "react-hook-form";
+
+import { Button } from "@cooper/ui/button";
+import { FormControl, FormField, FormItem } from "@cooper/ui/form";
+import { useCallback } from "react";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectSeparator,
+  SelectTrigger,
+  SelectValue,
+} from "@cooper/ui/select";
+import { Industry } from "@cooper/db/schema";
+import type { IndustryType, LocationType } from "@cooper/db/schema";
+import { api } from "~/trpc/react";
+import { usePathname, useRouter } from "next/navigation";
+import { z } from "zod";
+
+interface SearchBarProps {
+  industry?: IndustryType;
+  location?: LocationType;
+}
+
+const _formSchema = z.object({
+  searchIndustry: z.string().optional(),
+  searchLocation: z.string().optional(),
+});
+type FormSchema = z.infer<typeof _formSchema>;
+
+export function CompanySearchBar({ industry, location }: SearchBarProps) {
+  const form = useFormContext();
+  const { handleSubmit, setValue } = form;
+  const router = useRouter();
+  const pathName = usePathname();
+
+  const [selectedIndustry, setSelectedIndustry] = useState<string | undefined>(
+    industry,
+  );
+  const [selectedLocation, setSelectedLocation] = useState<string | undefined>(
+    location?.city,
+  );
+  const { data: locations = [] } = api.location.list.useQuery();
```

Comment:
I think this is causing the extreme latency which I'm seeing on the preview deployment. I think Tracy has solved this problem though. Spend some time looking through `review-section.tsx` in the form to see how Tracy implemented a Locations combobox. 
This should help with the problems we are facing right now. Once we get that figured out I can review (it is too slow to test right now)